docs/api/api.yml

```diff
@@ -386,29 +386,32 @@ paths:
     post:
       tags:
       - directory
-      summary: Adds a file or a subdirectory.
+      summary: Adds one or more files.
       description: |
-        Sub request of a request with `Content-Type` value `multipart/form-data`.
-        By default the files listed in the body of the main request are saved in `dir_path` under the file name provided
-        in the `Content-Disposition` header. However, if a file has the extension ZIP, it
-        is extracted in `dir_path` and `Content-Disposition` is ignored (and may be left out).
-        If `dir_path`, or any of its parents do not exist yet, they are first created.
+        The `Content-Type` must always be `multipart/form-data`, which means the body is formatted using the
+        [multipart media type](https://tools.ietf.org/html/rfc2046.html#section-5.1); each part contains one file.
+        The header area of each part specifies the file name using a `Content-Disposition` header.
+
+        If the body contains only one part and that part has `Content-Type: application/zip`, then the body
+        is a file in ZIP format. This ZIP file can contain a directory hierarchy. The ZIP file will be
+        unzipped in `dir_path`. For each file in the ZIP file, the path in the deposit will be `dir_path`/`path_in_zip`.
+
+        In all other cases the files are all stored in the directory pointed to by `dir_path` and it is not
+        allowed to include any files with `Content-Type: application/zip`. If multiple ZIP Note, that this means that only a flat
+        list of files can be sent this way, i.e. not including any directory hierarchy.
       parameters:
       - $ref: "#/components/parameters/DepositId"
       - $ref: "#/components/parameters/DirPath"
       - $ref: "#/components/parameters/ContentDisposition"
       requestBody:
         required: true
         content:
-          application/octet-stream:
-            schema:
-              # a binary file of any type
-              type: string
-              format: binary
-          application/zip:
+          multipart/form-data:
             schema:
-              type: string
-              format: binary
+              type: array
+              items:
+                type: string
+                format: binary
       responses:
         201:
           description: file or subdirectory added
```

Comment:
the 'adding a subdirectory' was dropped from the description. is that still possible?